lms/djangoapps/courseware/tests/test_microsites.py

```diff
@@ -129,6 +133,26 @@ def test_no_configuration_anonymous_homepage_content(self):
         # assert that footer template has been properly overriden on homepage
         self.assertNotContains(resp, 'This is a Test Site footer')
 
+        # The number of courses on home page is not limited by default,
+        # so the "View all Courses" link should not be visible.
+        self.assertNotContains(resp, 'View all Courses')
+
+    @override_settings(SITE_NAME=settings.MICROSITE_TEST_HOSTNAME)
+    def test_site_homepage_course_max(self):
+        # Add another course to the site.
+        CourseFactory.create(
+            display_name='Another Course',
+            org='TestSiteX',
+            emit_signals=True,
+        )
+
+        resp = self.client.get('/', HTTP_HOST=settings.MICROSITE_TEST_HOSTNAME)
+        self.assertEqual(resp.status_code, 200)
+
+        # Site now has three visible courses, but the HOMEPAGE_COURES_MAX setting is set to 2,
+        # so we should see a link to all courses.
+        self.assertContains(resp, 'View all Courses')
```

Comment:
Could you also test the case where HOMEPAGE_COURES_MAX exceeds the number of courses?